pkg/controller/pod_control.go

```diff
@@ -158,6 +165,9 @@ func (c *realPodControl) UpdateMetaInfo(tc *v1alpha1.TidbCluster, pod *corev1.Po
 			}
 			for _, store := range stores.Stores {
 				addr := store.Store.GetAddress()
+
+				// It's fine when ClusterDomain is annotationed, because the instance out of the cluster is ruled out of Stores and listed in PeerStores.
```

Comment:
```suggestion
				// It's fine when spec.ClusterDomain is set, because the instances out of this cluster are filtered out of Stores and listed in PeerStores.
```